src/backend/listingAction.js

```diff
@@ -0,0 +1,89 @@
+const p = require('bluebird');
+const databaseConnector = require('./databaseConnector');
+const User = require('./user');
+const functionCallback = (func, callback) => {
```

Comment:
Będę uczył o promisach, obiecuję. ;)

Przerabiasz tu promisa na callback tylko dlatego, że nie wyjaśniłem dobrze koncepcji, że wystarczy zwrócić referencję na promise z funkcji i używać go dalej - na zewnątrz. Zachowujesz ciągłość łańcucha kolejnych kroków i propagację błędów, dzię…i czemu przypięcie error handlera na koniec jest prostsze i daje lepsze efekty (gwarancja, że wszystkie błędy trafią do error handlera)

We wszystkich apkach na produkcji od ponad dwóch lat nie mieliśmy ani jednego niezłapanego błędu z naszego kodu właśnie dzięki temu. 
W ogóle jedyny niezłapany błąd w historii pochodził z asynchronicznego wywołania w bibliotece do zapytań http, którego nie dało się złapać z zewnątrz (bo ma interfejs na callbackach i sama nie złapała tego błędu)